shotover-proxy/src/message/mod.rs

```diff
@@ -18,7 +18,10 @@ pub type Messages = Vec<Message>;
 pub struct Message {
     pub details: MessageDetails,
     pub modified: bool,
+    /// The frame in the format defined by the protocol.
     pub original: RawFrame,
+    /// If return_to_sender true then the message is sent back to the originating system.
+    pub return_to_sender: bool,
```

Comment:
Is `return_to_sender` only used when there is a protocol error?

I think if a Message is an Protocol error there should be a variant in `MessageDetails` representing this. We don't want to get in the habit of adding boolean flags for every different type of Message.

@rukai & @benbromhead any thoughts on the right abstraction for this?

Reply:
MessageDetails shouldn't need to know about protocol errors (It should be protocol unaware... even if it knows the query language etc)

Reply:
I think MessageDetails::ReturnToSender does make sense!
If we are returning it directly to the sender then there is no need to have the generic contents of the message provided by the other MessageDetails variants.